tests/test_run_service.py

```diff
@@ -0,0 +1,265 @@
+        logger.addHandler(file_handler)
+    
+    return logger
+
+def test_run_service():
+    """Test running the run_service.py script using pexpect."""
+    
+    # Cleanup previous run artifacts
+    cleanup_previous_run()
+    
+    # Setup logging
+    timestamp = datetime.now().strftime('%Y%m%d_%H%M%S')
+    log_file = Path(f'test_run_service_{timestamp}.log')
+    logger = setup_logging(log_file)
+    
+    logger.info("Starting run_service.py test")
+    logger.info(f"Log file: logs/{log_file}")
+
+    
+    try:
+        script_path = Path("./run_service.sh")
+        os.chmod(script_path, 0o755)
+
+        # Start the process with pexpect
+        child = pexpect.spawn(
+            'bash ./run_service.sh',  # Use bash explicitly
+            encoding='utf-8',
+            timeout=600,
+            cwd="."
+        )
+        
+        # Enable logging of the interaction
+        child.logfile = sys.stdout
+        
+        # Handle the interaction
+        while True:
+            try:
+                patterns = list(PROMPTS.keys())
+                index = child.expect(patterns, timeout=600)
+                pattern = patterns[index]
+                response = PROMPTS[pattern]
+                
+                # Log the interaction
+                logger.info(f"Matched prompt: {pattern}", extra={'is_expect': True})
+                
+                # Handle response based on type
+                if callable(response):
+                    output = child.before + child.after
+                    response = response(output, logger)
+                
+                if "password" in pattern.lower():
+                    logger.info("Sending: [HIDDEN]", extra={'is_input': True})
+                else:
+                    logger.info(f"Sending: {response}", extra={'is_input': True})
+                
+                # Send the response
+                child.sendline(response)
+                
+            except pexpect.EOF:
+                logger.info("Process completed")
+                break
+            except pexpect.TIMEOUT:
+                logger.error("Timeout waiting for prompt")
+                break
+            except Exception as e:
+                logger.error(f"Error handling prompt: {str(e)}")
+                break
+        
+        # Get the exit status
+        child.close()
+        if child.exitstatus == 0:
+            logger.info("Test completed successfully")
+            # Check Docker status
+            if not check_docker_status(logger):
+                raise Exception("Docker container check failed")
+                
+            # Check service health
+            if not check_service_health(logger):
+                raise Exception("Service health check failed")
+                
+            logger.info("All checks passed successfully")
+        else:
+            logger.error(f"Test failed with exit status {child.exitstatus}")
+            
```

Comment:
this should fail the test, right?
```suggestion
            raise Exception(f"Test failed with exit status {child.exitstatus}")
            
```